modules/dnn/src/dnn.cpp

```diff
@@ -2639,6 +2641,22 @@ struct Net::Impl
 
                         if(inp_i_data->skip || inp_i_data->consumers.size() != 1)
                             break;
+#ifdef HAVE_CUDA
+                        if (preferableBackend == DNN_BACKEND_CUDA &&
+                            (inp_i_data->layerInstance->supportBackend(DNN_BACKEND_CUDA) == false || inp_i_data->layerInstance->type == "Region"))
```

Comment:
Here is an opportunity for a future regression. If somebody adds a layer which computes the final output on CPU (like region layer) but they forget to add an entry here and also does not test what would happen if their layer appears before a concat (who would even think about a test like this), this will throw off assertions in `CUDABackendWrapper`.
Possible solutions:
- have a test which would connect every layer that exists in OCV DNN to a concat
- use a whitelist instead of a blacklist
  - allow concat fusion for few types of input layers like Convolution, Pooling, etc. (this is what OCL is doing currently)

Reply:
let's use whitelist. Stability has higher priority than performance optimization for rare cases.